packages/scandipwa/src/component/CheckoutAgreements/CheckoutAgreements.component.js

```diff
@@ -0,0 +1,148 @@
+/**
+ * ScandiPWA - Progressive Web App for Magento
+ *
+ * Copyright © Scandiweb, Inc. All rights reserved.
+ * See LICENSE for license details.
+ *
+ * @license OSL-3.0 (Open Software License ("OSL") v. 3.0)
+ * @package scandipwa/base-theme
+ * @link https://github.com/scandipwa/base-theme
+ */
+
+import PropTypes from 'prop-types';
+import { PureComponent } from 'react';
+
+import Field from 'Component/Field';
+
+import './CheckoutAgreements.style.scss';
+
+/**
+ * @class CheckoutAgreements
+ * @namespace Component/CheckoutAgreements/Component
+ */
+export class CheckoutAgreements extends PureComponent {
+    static propTypes = {
+        termsAreEnabled: PropTypes.bool.isRequired,
+        termsAndConditions: PropTypes.array.isRequired,
+        setAgreementsStatus: PropTypes.func.isRequired,
+        showAgreementPopup: PropTypes.func.isRequired
+    };
+
+    state = {
+        isTermsAndConditionsAccepted: false,
+        selectedAgreementsList: [],
+        requiredAgreementsList: []
+    };
+
+    componentDidMount() {
+        const { termsAreEnabled } = this.props;
+
+        if (termsAreEnabled) {
+            this.setRequiredAgreements();
+        } else {
+            this.setState({ isTermsAndConditionsAccepted: true });
+        }
+    }
+
+    setRequiredAgreements = () => {
+        const { termsAndConditions } = this.props;
+        const requiredAgreementsList = [];
+
+        termsAndConditions.map(({ agreement_id }) => requiredAgreementsList.push(agreement_id));
+        this.setState({ requiredAgreementsList });
+    };
+
+    checkTermsAndConditions = () => {
+        const { setAgreementsStatus } = this.props;
+        const { requiredAgreementsList, selectedAgreementsList } = this.state;
+        const allRequiredSelected = requiredAgreementsList.every((elem) => selectedAgreementsList.indexOf(elem) > -1);
+
+        setAgreementsStatus(allRequiredSelected);
+        this.setState({ isTermsAndConditionsAccepted: allRequiredSelected });
+    };
+
+    updateAgreementsList = (agreementId, isChecked) => {
+        const { selectedAgreementsList } = this.state;
+
+        if (isChecked) {
+            const agreements = selectedAgreementsList.concat(+agreementId);
```

Comment:
Usage of `+${value}` to convert string to number is kind of a hack way of doing this conversion, can you please use standard `Number.parseInt` for that?